internal/commands/exec/exec.go

```diff
@@ -206,10 +270,47 @@ func RunExec(config ExecConfig) (ret int) {
 	return 0
 }
 
-func login(config ExecConfig) (*api.Client, string, error) {
+// The state contains the server address and ACL token (if applicable).
+func makeClient(config ExecConfig, s discovery.State) (*api.Client, error) {
+	config.ConsulConfig.Address = fmt.Sprintf("%s:%d", s.Address.IP.String(), config.ConsulHTTPPort)
+	// config.ConsulConfig.Address = fmt.Sprintf("%s:%d", s.Address, config.ConsulHTTPPort)
+	if s.Token != "" {
+		config.ConsulConfig.Token = s.Token
+	}
+	// config.ConsulConfig.Token = s.Token
+
+	// TODO: is this necessary?
+	if config.ConsulConfig.Transport == nil {
+		tlsClientConfig, err := api.SetupTLSConfig(&config.ConsulConfig.TLSConfig)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to configure TLS transport: %w", err)
+		}
+
+		config.ConsulConfig.Transport = &http.Transport{TLSClientConfig: tlsClientConfig}
+	} else if config.ConsulConfig.Transport.TLSClientConfig == nil {
+		tlsClientConfig, err := api.SetupTLSConfig(&config.ConsulConfig.TLSConfig)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to configure TLS transport: %w", err)
+		}
+
+		config.ConsulConfig.Transport.TLSClientConfig = tlsClientConfig
+	}
+	config.ConsulConfig.HttpClient.Transport = config.ConsulConfig.Transport
+
+	return api.NewClient(&config.ConsulConfig)
+}
+
+func login(config ExecConfig, s discovery.State) (*api.Client, string, error) {
```

Comment:
TODO: login/logout functionality appears to be supported directly within the consul-server-discovery library